### PR TITLE
made the existing test_vrf, test_vrf_af working for both nexus and XR

### PR DESCRIPTION
**Pull request by HelenHZhang**
Rally story US63947 (TA88173, TA88174).
Both tests passed on nexus and xr.

### CODE REVIEW

File: tests/beaker_tests/cisco_vrf/test_vrf.rb [hunk 1]
```diff
@@ -108,104 +43,47 @@
...
-  # Workaround for (ioctl) facter bug on n7k ***
-  tests[id][:code] = [0, 2] if platform[/n7k/]
-
-  # Build the manifest for this test
-  build_manifest_vrf(tests, id)
-
-  test_harness_common(tests, id)
-  tests[id][:ensure] = nil
+def unsupported_properties(_tests, _id)
+  return [] if operating_system == 'nexus'
```
Comment (glennmatthews) [inline tests/beaker_tests/cisco_vrf/test_vrf.rb]: In the old code above, it looks like route_distinguisher is unsupported except on n7k/n9k and vni is unsupported except on n9k. Am I missing where that check is now enforced?

**Comment by chrisvanheuveln** [comment]
:+1: 

Note: I found some additional dependencies for `rd`, `route_target*` and `vni` on the n7k but I can fix those up in another PR.

File: tests/beaker_tests/cisco_vrf/test_vrf.rb (hunk 1 above)
Comment (chrisvanheuveln) [inline tests/beaker_tests/cisco_vrf/test_vrf.rb]: @glennmatthews Yes, those checks are missing but I found some add'l dependencies on N7k anyway. I was going to add my own fixes but they will be easier to do with Helen's commit in place; ie. I'd prefer to merge her code as is and I'll follow up.

Comment (HelenHZhang) [inline tests/beaker_tests/cisco_vrf/test_vrf.rb]: @glennmatthews
I am trying keeping the original logic.

**Comment by glennmatthews** [comment]
:+1: with the understanding that Chris will refactor for NX as needed.

**Comment by glennmatthews** [comment]
@HelenHZhang Are any updates to demo manifests or README.md needed?

File: README.md [hunk 2]
```diff
@@ -2355,6 +2355,14 @@ device.
 | N7k      | unsupported        | unsupported            |
 | IOS XR   | TODO               | TODO                   |
 
+#### <a name="cisco_vrf-caveats">Caveats</a>
```
Comment (glennmatthews) [inline README.md]: These caveat tables need to be linked from the table under [Provider Support Across Platforms](https://github.com/cisco/cisco-network-puppet-module#provider-support-across-platforms)

Comment (HelenHZhang) [inline README.md]: added. I need to make a note to update README.md, and demo_xxx.rb

**Pull request merged**